### ISSUE TITLE
can't use the docker fluentd log driver when use unix style

### ISSUE THREAD

**Issue by mercykevin** [moby/moby#33350]
I run this command from cli

` docker run --log-driver=fluentd --log-opt fluentd-address=unix:///srv/docker/fluentd/fluent.sock redis:3.2.9`

but I get this error ,maybe when the docker get fluentd-address option it loss unix://prefix
`Error response from daemon: invalid fluentd-address unix:///srv/docker/fluentd/fluent.sock: strconv.ParseInt: parsing "///srv/docker/fluentd/fluent.sock": invalid syntax.`

**Comment by ripcurld0** [comment]
@mercykevin can you also provide details on how you setup fluentd?
I would like to reproduce this problem.

**Comment by cpuguy83** [comment]
Can you provide all the details from the issue template?
Namely the output of `docker version` and `docker info`?

Thanks.

**Comment by cpuguy83** [comment]
Note that support for transport schemes was added only in 1.13+.

**Comment by mercykevin** [comment]
@cpuguy83 
Thank you very much.

**Description**
can't use unix socket "unix://" protocol with docker fluentd driver.

**Steps to reproduce the issue:**
1. run fluentd ,use  "unix socket " style as log event source 
2. run docker container with fluentd driver, the log option is  "--fluentd-address=unix:///srv/docker/fluentd/fluent.sock"
command 
```
docker run --log-driver=fluentd --log-opt fluentd-address=unix:///srv/docker/fluentd/fluent.sock redis:3.2.9
```
3. 

**Describe the results you received:**


**Describe the results you expected:**
I have got this error:
Error response from daemon: invalid fluentd-address unix:///srv/docker/fluentd/fluent.sock: strconv.ParseInt: parsing "///srv/docker/fluentd/fluent.sock": invalid syntax.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   7392c3b/1.12.6
 Built:        Fri Jan  6 22:16:21 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   7392c3b/1.12.6
 Built:        Fri Jan  6 22:16:21 2017
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 32
 Running: 22
 Paused: 0
 Stopped: 10
Images: 64
Server Version: 1.12.6
Storage Driver: devicemapper
 Pool Name: docker-202:1-529557-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 12.97 GB
 Data Space Total: 107.4 GB
 Data Space Available: 94.41 GB
 Metadata Space Used: 28.65 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.119 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.93-RHEL7 (2015-01-28)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null bridge host overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options:
Kernel Version: 4.4.35-33.55.amzn1.x86_64
Operating System: Amazon Linux AMI 2016.09
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.956 GiB
Name: web0003
ID: D7O6:U4ZC:JCWF:FEBW:AYHY:5TI3:6QFU:JYBK:VNOG:ZHTE:EBMN:DQ5O
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Insecure Registries:
 docker-inner:5000
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
AWS

**Comment by thaJeztah** [comment]
Yes, that's expected - support for unix:// sockets was added in docker 1.13 (as @cpuguy83 explained) - see https://github.com/moby/moby/pull/26088

You look to be running an older version of docker that does not yet support that feature.

I'll close this issue because of the above, but feel free to continue the conversation
